sdcm/cluster.py

```diff
@@ -2432,25 +2449,29 @@ def get_scylla_args(self):
         return self.params.get('append_scylla_args_oracle') if self.name.find('oracle') > 0 else \
             self.params.get('append_scylla_args')
 
-    def get_seed_nodes_from_node(self):
-        if self.seed_nodes_ips is None:
-            node = self.nodes[0]  # pylint: disable=no-member
-            yaml_dst_path = os.path.join(tempfile.mkdtemp(prefix='sct'),
-                                         'scylla.yaml')
-            node.remoter.receive_files(src=SCYLLA_YAML_PATH,
-                                       dst=yaml_dst_path)
-            with open(yaml_dst_path, 'r') as yaml_stream:
-                conf_dict = yaml.safe_load(yaml_stream)
-                try:
-                    self.seed_nodes_ips = conf_dict['seed_provider'][0]['parameters'][0]['seeds'].split(',')
-                except Exception as details:
-                    self.log.debug('Loaded YAML data structure: %s', conf_dict)  # pylint: disable=no-member
-                    self.log.error('Scylla YAML config contents:')  # pylint: disable=no-member
-                    with open(yaml_dst_path, 'r') as yaml_stream:
-                        self.log.error(yaml_stream.read())  # pylint: disable=no-member
-                    raise ValueError('Exception determining seed node ips: %s' %
-                                     details)
-        return self.seed_nodes_ips
+    def get_seed_nodes(self):
+        seed_nodes = []
+        for node in self.nodes:  # pylint: disable=no-member
+            if node.ip_address in self.seed_nodes_ips:
+                seed_nodes.append(node)
+        assert seed_nodes, "We should have at least one selected seed by now"
+        return seed_nodes
+
+    def get_non_seed_nodes(self):
+        assert self.seed_nodes_ips, "We should have at least one selected seed by now"
+        return [node for node in self.nodes if not node.is_seed]  # pylint: disable=no-member
+
+    def get_seed_nodes_ips(self):
```

Comment:
this is actually one liner that can be used in the relevant function, no need to create a method for it

Reply:
Yes, I know. I did it because we do it in the different places.

Reply:
done

Reply:
I still see it in the code